### DISCUSSION TITLE
Is it possible to make few route prefixes with one locale?

### DISCUSSION THREAD

**Discussion by Sin1tar** [i18next/next-i18next#1649] [Q&A] [answered]
Hello everyone

Are there other ways to get route aliases?

want have this:
```
/en-US -> public\locales\en\...
/en-CA -> public\locales\en\...
/en-BR -> public\locales\en...
```

but right now i can do it only with CTRL-C - CTRL-V:
```
/en-US -> public\locales\en-US\...
/en-CA -> public\locales\en-CA\...
/en-BR -> public\locales\en-BR\...
```

thank you so much!

**Comment by isaachinman** [comment, marked as answer]
This can be achieved via [fallback languages](https://www.i18next.com/principles/fallback#fallback-to-different-languages), through i18next.

**Reply by Sin1tar** [reply]
Thanks for the answer, right now I've read the documentation, added the necessary property to the object, but I get an error 404

```
next-i18next.config.js

module.exports = {
  i18n: {
    defaultLocale: 'en',
    locales: ['en', 'it'],
    localePath: path.resolve('./public/locales'),
    reloadOnPrerender: true,
    fallbackLng: {
      'en-US': ['en'],
    }
  },
}
```

**Reply by isaachinman** [reply]
The `fallbackLng` option should be at the top level of your `next-i18next.config`, not nested inside `i18n`.

**Reply by Sin1tar** [reply]
Thank you so much!

**Reply by Sin1tar** [reply]
> The `fallbackLng` option should be at the top level of your `next-i18next.config`, not nested inside `i18n`.

uh, again 404..

```
module.exports = {
  fallbackLng: {
    'en-CA': ['en'],
    'default': ['en'],
  },
  cleanCode: true,
  i18n: {
    defaultLocale: 'en',
    locales: [
      'en-CA',
      'en',
      'it',
    ],
  },
```
}

http://localhost:3000/en OK
http://localhost:3000/en-CA 404

**Reply by isaachinman** [reply]
@SergeyMustafin Can you link a reproducible repository?

**Reply by Sin1tar** [reply]
> @SergeyMustafin Can you link a reproducible repository?

sure, please let me few minutes

**Reply by Sin1tar** [reply]
> @SergeyMustafin Can you link a reproducible repository?

https://codesandbox.io/s/nervous-hopper-kvcz3

**Reply by isaachinman** [reply]
Please share the project as a public GitHub repository.

**Reply by Sin1tar** [reply]
> Please share the project as a public GitHub repository.

https://github.com/SergeyMustafin/nexti18n-fallbackLng

**Reply by isaachinman** [reply]
Thanks. Is this the error you are seeing, or is it something else?


![Screenshot 2022-01-25 at 17 49 33](https://user-images.githubusercontent.com/10575782/151031402-06dc5648-6d29-4636-9b5e-52f6cd6173b9.png)

**Reply by Sin1tar** [reply]
> Thanks. Is this the error you are seeing, or is it something else?
> 
> ![Screenshot 2022-01-25 at 17 49 33](https://user-images.githubusercontent.com/10575782/151031402-06dc5648-6d29-4636-9b5e-52f6cd6173b9.png)

Yes, only this error.
If i extract 'en-CA' from 'locales', then this error will fix.
But, also 'en-CA' declared in fallbackLng (i expect what this will show locale by route but have only 404)

is it possible to have few routes from one locale?

need:
http://localhost:3000/en-CA (translation from public\locales\en)
http://localhost:3000/en-DE (translation from public\locales\en)
http://localhost:3000/en (translation from public\locales\en)
etc

now:
http://localhost:3000/en-CA (translation from public\locales\en-CA) 404
http://localhost:3000/en (translation from public\locales\en) OK

**Reply by isaachinman** [reply]
I think we actually need to update the logic in [`createConfig`](https://github.com/isaachinman/next-i18next/blob/master/src/config/createConfig.ts) to not check locales via the fs when they are covered with a fallback.

@SergeyMustafin Do you mind creating an actual issue for this? Thanks very much for your help in debugging.

**Reply by Sin1tar** [reply]
> I think we actually need to update the logic in [`createConfig`](https://github.com/isaachinman/next-i18next/blob/master/src/config/createConfig.ts) to not check locales via the fs when they are covered with a fallback.
> 
> @SergeyMustafin Do you mind creating an actual issue for this? Thanks very much for your help in debugging.

here is: https://github.com/isaachinman/next-i18next/issues/1652
Thank you so much!
